src/plugins/XHRUpload.js

```diff
@@ -90,6 +91,19 @@ module.exports = class XHRUpload extends Plugin {
       formPost.append(item, file.meta[item])
     })
 
+    let data = opts.data
+    if (typeof opts.data === 'function') {
+      data = opts.data(file)
+    }
+
+    if (data != null) {
+      (typeof data === 'object')
+        ? Object.keys(data).forEach((key) => {
+          formPost.append(key, data[key])
+        })
```

Comment:
What's wrong with a if here? Also, `Object.entries` is more appropriate if you need both the key and the value.
```js
Object.entries(data).forEach(entry => formPost.append(...entry))
```
You will probably want to deal with nested objects somehow (probably worth throwing since it's really an invalid behavior and will result in `"[object Object]"` being sent)

Reply:
I don't understand what you mean by "What's wrong with a if here?"?

Reply:
We use `Object.keys` everywhere because `Object.entries` is not supported in some browsers. I would also prefer a regular `if` statement here!

Reply:
Ah, babel will be the bane of me. 😇